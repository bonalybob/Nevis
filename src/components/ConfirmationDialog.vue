<template>
  <div
    class="fixed top-0 pt-8 flex w-full h-full bg-blue-point4 flex-col justify-around content-center select-none"
  >
    <div
      class="-mt-16 bg-white justify-around content-center flex-none my-shadow-lg mx-12 sm:mx-16 md:mx-20 px-3 pt-2 pb-3 text-center select-none"
    >
      <h1>{{ heading }}</h1>
      <p class="my-1 select-none">{{ message }}</p>
      <button
        @click="confirmAction(false)"
        class="w-1/3 sm:w-1/4 m-1 mt-2 py-1 px-3 text-blue font-heading border border-solid border-blue hover:bg-blue-point4"
      >
        {{ cancel }}
      </button>
      <button
        @click="confirmAction(true)"
        class="w-1/3 sm:w-1/4 m-1 mt-2 py-1 px-3 bg-blue text-white font-heading hover:bg-blue-point8 border border-solid border-blue"
      >
        {{ confirm }}
      </button>
    </div>
  </div>
</template>

<script>
export default {
  name: 'ConfirmationDialog',

  props: {
    heading: {
      type: String,
      default: '',
    },
    message: {
      type: String,
      default: '',
    },
    confirm: {
      type: String,
      default: '',
    },
    cancel: {
      type: String,
      default: 'Cancel',
    },
  },

  methods: {
    confirmAction: function(value) {
      this.$emit('close', value)
    },
  },
}
</script>
