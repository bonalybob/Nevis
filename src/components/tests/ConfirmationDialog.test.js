import { mount } from '@vue/test-utils'
import ConfirmationDialog from '@/components/ConfirmationDialog'

test('Is a Vue Instance', () => {
  const wrapper = mount(ConfirmationDialog, {
    stubs: ['router-link'],
  })
  expect(wrapper.isVueInstance()).toBeTruthy()
})

test('Renders Correctly', () => {
  const wrapper = mount(ConfirmationDialog, {
    stubs: ['router-link'],
  })
  wrapper.setProps({
    title: 'Pop-up',
    message: 'Hello',
    cancel: 'Cancel',
    confirm: 'Hello',
  })
  expect(wrapper.element).toMatchSnapshot()
})

test('Emits Value on Close', () => {
  const wrapper = mount(ConfirmationDialog, {
    stubs: ['router-link'],
  })
  wrapper.vm.confirmAction('Hello')
  wrapper.vm.confirmAction(123)
  expect(wrapper.emitted().close.length).toBe(2)
  expect(wrapper.emitted().close[0]).toEqual(['Hello'])
  expect(wrapper.emitted().close[1]).toEqual([123])
})
