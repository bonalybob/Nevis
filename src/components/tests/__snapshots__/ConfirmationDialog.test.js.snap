// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Correctly 1`] = `
<div
  class="fixed top-0 pt-8 flex w-full h-full bg-blue-point4 flex-col justify-around content-center select-none"
>
  <div
    class="-mt-16 bg-white justify-around content-center flex-none shadow-md mx-12 sm:mx-16 md:mx-20 px-3 pt-2 pb-3 text-center select-none"
  >
    <h1>
      
    </h1>
     
    <p
      class="my-1 select-none"
    >
      Hello
    </p>
     
    <button
      class="w-1/3 sm:w-1/4 m-1 mt-2 py-1 px-3 text-blue font-heading border border-solid border-blue hover:bg-blue-point4"
    >
      
      Cancel
    
    </button>
     
    <button
      class="w-1/3 sm:w-1/4 m-1 mt-2 py-1 px-3 bg-blue text-white font-heading hover:bg-blue-point8 border border-solid border-blue"
    >
      
      Hello
    
    </button>
  </div>
</div>
`;
